### ISSUE TITLE
Groups of tasks - constraints question

### ISSUE THREAD

**Issue by ollamh** [timnon/pyschedule#77]
Hello @timnon !

I have a use case where I create iteratively a large set of tasks, some of which can be part of a group, some are not. After that I set constrains (mostly LAX and TIGHT ones).  The main issue is that I have to set LAX constraints for ALL tasks in a group ne by one, due to the fact that I don't use `pyschedule.Tasks` class. Is there a way to "group" those tasks groups under one "pointer"?

Here is an example code:
```
       import pyschedule
       s = pyschedule.Scenario(‘test_periods’, horizon=100)
       t1 = s.Task(‘Graphics_1’, length=4, delay_cost=1 , group=‘1’)
       t2 = s.Task(‘Graphics_2’, length=4, delay_cost=1, group=‘1’)
       t3 = s.Task(‘Graphics_3’, length=4, delay_cost=1,group=‘1’)
       t4 = s.Task(‘Graphics_4’, length=4, delay_cost=1,group=‘1’)
       t5 = s.Task(‘Graphics_5’, length=4, delay_cost=1,group=‘1’)
       t6 = s.Task(‘Graphics_6’, length=4, delay_cost=1)
       t7 = s.Task(‘Graphics_7’, length=4, delay_cost=1)

       resource1 = s.Resource(‘R1’)
       resource2 = s.Resource(‘R2’)
       resource3 = s.Resource(‘R3’)
      # Here I have to set constraint for every task in a group...
       s += t6 > t5
       s += t6 > t4
       s += t6 > t3
       s += t6 > t2
       s += t6 > t1
       s += t7 > t6
       t1 += resource1
       t2 += resource1
       t3 += resource1
       t4 += resource1
       t5 += resource1
       t6 += resource2
       t7 += resource3

       pyschedule.solvers.mip.solve(s, kind=‘CBC’, msg=1, ratio_gap=1.2)
       print(s.solution())
       pyschedule.plotters.matplotlib.plot(
           s, resource_height=100.0)
```

Thanks!

**Comment by timnon** [comment]
The shortest way to describe your requirements is:
```
S = Scenario('test',horizon=10)

T = S.Tasks('T_',num=7,delay_cost=1)
R = S.Resources('R_',num=3)

S += T[5] > T[:5]
S += T[6] > T[5]

T[:5] += R[0]
T[5] += R[1]
T[6] += R[2]

solvers.mip.solve(S,msg=1,kind='CBC')
plotters.matplotlib.plot(S)
```
Since your first 5 tasks are completely interchangeable, you could also define them as a separate group and set `is_task=True`. This will tell the solver that these tasks can be interchanged on a schedule, which reduces the solving time. However, this is strictly an additional meta-information for tuning and not necessary to define a scenario.

**Comment by timnon** [comment]
Please update to the newest version

**Comment by ollamh** [comment]
Thank you @timnon for the response, I'm already using version 0.2.34. 
The solution proposed by you does not suit well my needs, I have list of `Task` object which have the same `group` attribute value, but they can be of a different length (I split one big task manually over a complex working timetable, including possible extra working hours or maintenance periods). 
I looked into pyschedule implementation and I noticed that it is inherited from the `_List` class, so, I made my class, so I can add any number of tasks into the task group.
